tracer/src/Datadog.Trace/Baggage.cs

```diff
@@ -0,0 +1,357 @@
+        }
+    }
+
+    void ICollection<KeyValuePair<string, string>>.Add(KeyValuePair<string, string> item)
+    {
+        Add(item.Key, item.Value);
+    }
+
+    /// <summary>
+    /// Removes the baggage value associated with the specified name.
+    /// </summary>
+    /// <param name="name">The baggage item name.</param>
+    /// <returns><c>true</c> if the object was removed successfully; otherwise, <c>false</c>.</returns>
+    public bool Remove(string name)
+    {
+        if (name == null!)
+        {
+            ThrowHelper.ThrowArgumentNullException(nameof(name));
+        }
+
+        if (_items is { Count: > 0 } list)
+        {
+            lock (list)
+            {
+                for (int i = 0; i < list.Count; i++)
+                {
+                    if (list[i].Key == name)
+                    {
+                        list.RemoveAt(i);
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    public bool Contains(KeyValuePair<string, string> item)
+    {
+        throw new System.NotImplementedException();
+    }
+
+    public void CopyTo(KeyValuePair<string, string>[] array, int arrayIndex)
+    {
+        throw new System.NotImplementedException();
+    }
+
+    /// <summary>
+    /// Gets the baggage value associated with the specified name.
+    /// </summary>
+    /// <param name="name">The baggage item name.</param>
+    /// <returns>Returns the baggage item value, or <c>null</c> if not found.</returns>
+    public string? Get(string name)
+    {
+        if (name == null!)
+        {
+            ThrowHelper.ThrowArgumentNullException(nameof(name));
+        }
+
+        return TryGetValue(name, out var value) ? value : null;
+    }
+
+    /// <summary>
+    /// Gets all baggage values.
+    /// </summary>
+    /// <returns>A new array that contains all baggage values.</returns>
+    public KeyValuePair<string, string>[] GetAll()
+    {
+        if (_items is { } list)
+        {
+            lock (list)
+            {
+                if (list.Count > 0)
+                {
+                    return _items.ToArray();
+                }
+            }
+        }
+
+        return [];
+    }
+
+    /// <summary>
+    /// Removes all baggage items.
+    /// </summary>
+    public void Clear()
+    {
+        if (_items is { } list)
+        {
+            lock (list)
+            {
+                _items?.Clear();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Adds the baggage items from <paramref name="baggage"/> to this baggage.
+    /// </summary>
+    public void Merge(Baggage? baggage)
+    {
+        if (baggage?.Count > 0)
+        {
+            var thisItems = EnsureListInitialized();
+            var otherItems = baggage._items!;
+
+            lock (thisItems)
+            {
+                lock (otherItems)
+                {
+                    thisItems.AddRange(otherItems);
```

Comment:
Should the `Merge` operation overwrite existing baggage items with the same key? This implementation so far only seems to append, leaving duplicate keys in the collection

Reply:
Oops, good catch!

Reply:
Fixed and added a test case.